### PR TITLE
Enable Wno-unused-private-field,Wunused-lambda-capture and fix CUDA warnings

### PR DESCRIPTION
**Pull request by cyyever**
This PR enables Wno-unused-private-field,Wunused-lambda-capture  and some CUDA warnings were fixed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -897,10 +897,8 @@ if(NOT MSVC)
   append_cxx_flag_if_supported("-Wno-error=inconsistent-missing-destructor-override" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Wconstant-conversion" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Wno-invalid-partial-specialization" CMAKE_CXX_FLAGS)
-  append_cxx_flag_if_supported("-Wno-unused-private-field" CMAKE_CXX_FLAGS)
```
Comment (albanD) [inline CMakeLists.txt:900]: Why are these removed from here?

Comment (cyyever) [reply CMakeLists.txt:900]: To identify and remove unused private field in torch. I think it a useful feature and if there is a third-party library requiring this fix, it should be added to its CMake target rather than enabled globally.

Comment (albanD) [reply CMakeLists.txt:900]: I think I'm just confused about the use of -W flag here. Doesn't removing this line actually removes the warning for the top level library?

Comment (cyyever) [reply CMakeLists.txt:900]: Removing it should apply to the whole library.

**Review by albanD — APPROVED** [review]
Sounds good!

**Review by malfet — APPROVED** [review]

File: test/edge/CMakeLists.txt [hunk 2]
```diff
@@ -4,6 +4,8 @@ set(TORCH_ROOT ${CMAKE_CURRENT_LIST_DIR}/../..)
 set(TEST_ROOT ${TORCH_ROOT}/test/edge)
 set(OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/out)
 file(GLOB_RECURSE all_python "${TORCH_ROOT}/torchgen/*.py")
+include(${TORCH_ROOT}/cmake/public/utils.cmake)
+append_cxx_flag_if_supported("-Wno-unused-private-field" CMAKE_CXX_FLAGS)
```
[the remark this answers is not in the document]
Comment (cyyever) [reply test/edge/CMakeLists.txt:8]: Many tests have unused private-fields. For example, I found 3 test locally. I plan to clean them soon so this line is a temporary patch to avoid breaking them now.
